modules/llm-cache/ds/kv_state_cache_block.cc

```diff
@@ -164,23 +167,21 @@ bool KVStateCacheBlockBuilder::IsFull() {
   return true;
 }
 
-void KVStateCacheBlockBuilder::Update(const KV_STATE_WITH_LAYER& kvState,
-                                      OffsetData* data) {
+void KVStateCacheBlockBuilder::Update(
+    const std::map<int, std::pair<LLMKV, LLMKV>>& kvState, OffsetData* data) {
   int index = this->FindEmptySlot();
   for (int currentLayer = 0; currentLayer < this->layer; currentLayer++) {
-    K_STATE keyState = (kvState.find(currentLayer)->second).first;
-    V_STATE valueState = (kvState.find(currentLayer)->second).second;
-    VINEYARD_ASSERT(keyState.length ==
-                    (size_t) this->dimension * sizeof(double));
-    VINEYARD_ASSERT(valueState.length ==
-                    (size_t) this->dimension * sizeof(double));
-
-    double* keyData = keyStateTensorBuilderList[currentLayer]->data();
-    double* valueData = valueStateTensorBuilderList[currentLayer]->data();
-    memcpy(keyData + index * this->dimension, keyState.data,
-           this->dimension * sizeof(double));
-    memcpy(valueData + index * this->dimension, valueState.data,
-           this->dimension * sizeof(double));
+    LLMKV keyState = (kvState.find(currentLayer)->second).first;
+    LLMKV valueState = (kvState.find(currentLayer)->second).second;
+    VINEYARD_ASSERT(keyState.length == (size_t) this->tensorBytes);
+    VINEYARD_ASSERT(valueState.length == (size_t) this->tensorBytes);
```

Comment:
Let `Update()` returns status and use `RETURN_ON_ERROR/RETUREN_ON_ASSERT`.

@vegetableysm remember to revise all usage of `VINEYARD_ASSERT` to see if it can be replaced by `RETUREN_ON_ASSERT`.

Reply:
Implemented in https://github.com/v6d-io/v6d/pull/1794/files#diff-58f18c2f80244c1a7accb540ac9d106b6f920e7679fee6591f6cc30d95018cadR168-R191